kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveGlobalConsumer.java

```diff
@@ -0,0 +1,398 @@
+ * forking Kafka Streams.</p>
+ */
+public class ResponsiveGlobalConsumer implements Consumer<byte[], byte[]> {
+
+  private final Consumer<byte[], byte[]> delegate;
+  private final Integer defaultApiTimeoutMs;
+  private final Admin admin;
+
+  public ResponsiveGlobalConsumer(
+      final Map<String, Object> config,
+      final Consumer<byte[], byte[]> delegate,
+      final Admin admin
+  ) {
+    final ConsumerConfig consumerConfig = new ConsumerConfig(config);
+
+    this.delegate = delegate;
+    this.defaultApiTimeoutMs = consumerConfig.getInt(ConsumerConfig.DEFAULT_API_TIMEOUT_MS_CONFIG);
+    this.admin = admin;
+  }
+
+  @Override
+  public Set<TopicPartition> assignment() {
+    return delegate.assignment();
+  }
+
+  @Override
+  public Set<String> subscription() {
+    return delegate.subscription();
+  }
+
+  @Override
+  public void subscribe(final Collection<String> topics, final ConsumerRebalanceListener listener) {
+    delegate.subscribe(topics, listener);
+  }
+
+  @Override
+  public void subscribe(final Collection<String> topics) {
+    delegate.subscribe(topics);
+  }
+
+  @Override
+  public void subscribe(final Pattern pattern, final ConsumerRebalanceListener listener) {
+    delegate.subscribe(pattern, listener);
+  }
+
+  @Override
+  public void subscribe(final Pattern pattern) {
+    delegate.subscribe(pattern);
+  }
+
+  @Override
+  public void unsubscribe() {
+    delegate.unsubscribe();
+  }
+
+  @Override
+  public void assign(final Collection<TopicPartition> partitions) {
+    subscribe(
+        partitions
+            .stream()
+            .map(TopicPartition::topic)
+            .collect(Collectors.toSet())
+    );
+  }
+
+  @Override
+  @Deprecated
+  public ConsumerRecords<byte[], byte[]> poll(final long timeoutMs) {
+    return poll(Duration.ofMillis(timeoutMs));
+  }
+
+  @Override
+  public ConsumerRecords<byte[], byte[]> poll(final Duration timeout) {
+    commitSync();
+    final ConsumerRecords<byte[], byte[]> poll = delegate.poll(timeout);
+    return SingletonConsumerRecords.of(poll);
+  }
+
+  @Override
+  public void commitSync() {
+    delegate.commitSync();
+  }
+
+  @Override
+  public void commitSync(final Duration timeout) {
+    delegate.commitSync(timeout);
+  }
+
+  @Override
+  public void commitSync(final Map<TopicPartition, OffsetAndMetadata> offsets) {
+    delegate.commitSync(offsets);
+  }
+
+  @Override
+  public void commitSync(final Map<TopicPartition, OffsetAndMetadata> offsets,
+      final Duration timeout) {
+    delegate.commitSync(offsets, timeout);
+  }
+
+  @Override
+  public void commitAsync() {
+    delegate.commitAsync();
+  }
+
+  @Override
+  public void commitAsync(final OffsetCommitCallback callback) {
+    delegate.commitAsync(callback);
+  }
+
+  @Override
+  public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets,
+      final OffsetCommitCallback callback) {
```

Comment:
super nit, but this is terrible formatting. Check your IDE settings! lol

Reply:
🤷 haha I think this actually is the "right" thing based on Google checkstyle conventions because otherwise it breaks 100 chars. if I were to manually do this I'd put them all on their own lines.